include/seqan3/alignment/matrix/detail/two_dimensional_matrix.hpp

```diff
@@ -321,13 +322,16 @@ class two_dimensional_matrix
  * data in a flattened one-dimensional vector.
  */
 template <typename value_t, typename allocator_t, matrix_major_order order>
-template <typename matrix_t>
+template <bool const_range>
 class two_dimensional_matrix<value_t, allocator_t, order>::basic_iterator :
-    public two_dimensional_matrix_iterator_base<basic_iterator<matrix_t>, order>
+    public two_dimensional_matrix_iterator_base<basic_iterator<const_range>, order>
 {
 private:
+    //!\brief Type of the parent range.
+    using parent_t = detail::maybe_const_range_t<const_range, two_dimensional_matrix>;
+
     //!\brief The base class type.
-    using base_t = two_dimensional_matrix_iterator_base<basic_iterator<matrix_t>, order>;
+    using base_t = two_dimensional_matrix_iterator_base<basic_iterator, order>;
```

Comment:
Why don't we need the template argument here any more?

Reply:
Because within the class body `basic_iterator` is actually the same as `basic_iterator<matrix_t>`. (It's the same as within the constructors)